yapapi/executor/__init__.py

```diff
@@ -498,6 +505,9 @@ async def worker_starter() -> None:
             for task in services:
                 if task is not process_invoices_job:
                     task.cancel()
+            if not activities_started:
+                # No need to wait for invoices
+                process_invoices_job.cancel()
```

Comment:
Wait. We have to process invoices for agreements that were signed — even if no activities related to those agreements were started.

Reply:
Good catch! Fixed in 2892a35197f4d7bafd6e06a3f28efbaec16a1dce